elixir-mode.el

```diff
@@ -273,6 +254,99 @@ for the Elixir programming language."
+the point where scan starts (defaults to current point), and LIMIT
+is used to limit the scan."
+  (let ((i 0))
+    (while (and (< i 3)
+                (or (not limit) (< (+ point i) limit))
+                (eq (char-after (+ point i)) quote-char))
+      (setq i (1+ i)))
+    i))
+
+(defun elixir-syntax-stringify ()
+  "Put `syntax-table' property correctly on single/triple quotes."
+  (let* ((num-quotes (length (match-string-no-properties 1)))
+         (ppss (prog2
+                   (backward-char num-quotes)
+                   (syntax-ppss)
+                 (forward-char num-quotes)))
+         (string-start (and (not (nth 4 ppss)) (nth 8 ppss)))
+         (quote-starting-pos (- (point) num-quotes))
+         (quote-ending-pos (point))
+         (num-closing-quotes
+          (and string-start
+               (elixir-syntax-count-quotes
+                (char-before) string-start quote-starting-pos))))
+    (cond ((and string-start (= num-closing-quotes 0))
+           ;; This set of quotes doesn't match the string starting
+           ;; kind. Do nothing.
+           nil)
+          ((not string-start)
+           ;; This set of quotes delimit the start of a string.
+           (put-text-property quote-starting-pos (1+ quote-starting-pos)
+                              'syntax-table (string-to-syntax "|")))
+          ((= num-quotes num-closing-quotes)
+           ;; This set of quotes delimit the end of a string.
+           (put-text-property (1- quote-ending-pos) quote-ending-pos
+                              'syntax-table (string-to-syntax "|")))
+          ((> num-quotes num-closing-quotes)
+           ;; This may only happen whenever a triple quote is closing
+           ;; a single quoted string. Add string delimiter syntax to
+           ;; all three quotes.
+           (put-text-property quote-starting-pos quote-ending-pos
+                              'syntax-table (string-to-syntax "|"))))))
+
+
+(defun elixir-syntax-propertize-interpolation ()
+  (let* ((beg (match-beginning 0))
+         (context (save-excursion (save-match-data (syntax-ppss beg)))))
+    (put-text-property beg (1+ beg) 'elixir-interpolation
+                       (cons (nth 3 context) (match-data)))))
+
+(defconst elixir-syntax-propertize-function
+  (syntax-propertize-rules
+   ((elixir-rx string-delimiter)
+    (0 (ignore (elixir-syntax-stringify))))
+   ((rx (group "#{" (0+ (not (any "}"))) "}"))
+    (0 (ignore (elixir-syntax-propertize-interpolation))))))
+
+(defun elixir-syntax-propertize-function (start end)
+  (let ((case-fold-search nil))
+    (goto-char start)
+    (funcall
+     (syntax-propertize-rules
+      ((elixir-rx string-delimiter)
+       (0 (ignore (elixir-syntax-stringify))))
+      ((rx (group "#{" (0+ (not (any "}"))) "}"))
+       (0 (ignore (elixir-syntax-propertize-interpolation)))))
+     start end)))
+
+;; (defun elixir-syntax-propertize-function (start end)
```

Comment:
All I meant was
https://github.com/elixir-lang/emacs-elixir/commit/bc2aa4f4f8fd4c96d2d8247b50d5568a9aff7a3e
... but no worries was simple enough to do myself. Thanks again I am sure
tons of users will be very happy at your improvements.

On Tue, May 26, 2015 at 2:11 PM, Andreas Röhler <notifications@github.com>
wrote:

> In elixir-mode.el
> <https://github.com/elixir-lang/emacs-elixir/pull/203#discussion_r31068707>
> :
>
> > +    (0 (ignore (elixir-syntax-stringify))))
> > +   ((rx (group "#{" (0+ (not (any "}"))) "}"))
> > +    (0 (ignore (elixir-syntax-propertize-interpolation))))))
> > +
> > +(defun elixir-syntax-propertize-function (start end)
> > +  (let ((case-fold-search nil))
> > +    (goto-char start)
> > +    (funcall
> > +     (syntax-propertize-rules
> > +      ((elixir-rx string-delimiter)
> > +       (0 (ignore (elixir-syntax-stringify))))
> > +      ((rx (group "#{" (0+ (not (any "}"))) "}"))
> > +       (0 (ignore (elixir-syntax-propertize-interpolation)))))
> > +     start end)))
> > +
> > +;; (defun elixir-syntax-propertize-function (start end)
>
> Am 26.05.2015 um 20:59 schrieb Matt DeBoard:
>  In elixir-mode.el <
> https://github.com/elixir-lang/emacs-elixir/pull/203#discussion_r31067476>:
> > + (0 (ignore (elixir-syntax-stringify)))) > + ((rx (group "#{" (0+ (not
> (any "}"))) "}")) > + (0 (ignore
> (elixir-syntax-propertize-interpolation)))))) > + > +(defun
> elixir-syntax-propertize-function (start end) > + (let ((case-fold-search
> nil)) > + (goto-char start) > + (funcall > + (syntax-propertize-rules > +
> ((elixir-rx string-delimiter) > + (0 (ignore (elixir-syntax-stringify)))) >
> + ((rx (group "#{" (0+ (not (any "}"))) "}")) > + (0 (ignore
> (elixir-syntax-propertize-interpolation))))) > + start end))) > + > +;;
> (defun elixir-syntax-propertize-function (start end) Not sure I understand?
> This is in the new PR (#203
> <https://github.com/elixir-lang/emacs-elixir/pull/203> <
> https://github.com/elixir-lang/emacs-elixir/pull/203>)
> Yes. Could you precise your question? First part of
> syntax-propertize-rules is the fix. Second part non-functional for the
> moment (interpolation is broken as announced) Will go on here after merge.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/emacs-elixir/pull/203/files#r31068707>.
>